dev-docs/bidder-adaptor.md

```diff
@@ -12,259 +11,504 @@ nav_section: adapters
+| `netRevenue` | Required                                    | Boolean defining whether the bid is Net or Gross. The value `true` is Net. Bidders responding with Gross-price bids should set this to false. | `false`                              |
+| `currency`   | Required                                    | 3-letter ISO 4217 code defining the currency of the bid.                                                                                      | `"EUR"`                              |
+| `vastUrl`    | Either this or `vastXml` required for video | URL where the VAST document can be retrieved when ready for display.                                                                          | `"http://vid.example.com/9876`       |
+| `vastXml`    | Either this or `vastUrl` required for video | XML for VAST document to be cached for later retrieval.                                                                                       | `<VAST version="3.0">...`            |
+| `dealId`     | Optional                                    | Deal ID                                                                                                                                       | `"123abc"`                           |
+
+### Register User Syncs
+
+All user ID sync activity must be done in one of two ways:
+
+1. The `getUserSyncs` callback of the `BaseAdapter` model
+2. The `userSync.registerSync` function
 
 {% highlight js %}
-var bidRequest  = utils.getBidRequest(id);
-var bidResponse = bidfactory.createBid(STATUS.NO_BID, bidRequest);
+
+{
+    getUserSyncs: function(syncOptions, serverResponses) {
+        const syncs = []
+        if (syncOptions.iframeEnabled) {
+            syncs.push({
+                type: 'iframe',
+                url: '//acdn.adnxs.com/ib/static/usersync/v3/async_usersync.html'
+            });
+        }
+        if (syncOptions.pixelEnabled && serverResponses.length > 0) {
+            syncs.push({
+                type: 'image',
+                url: serverResponses[0].body.userSync.url
+            });
+        }
+        return syncs;
+    }
+}
+
 {% endhighlight %}
 
-Example:
+## Supporting Video
+
+Follow the steps in this section to ensure that your adapter properly supports video.
+
+### Step 1: Register the adapter as supporting video
+
+Add the `supportedMediaTypes` argument to the spec object, and make sure `video` is in the list:
 
 {% highlight js %}
 
-// In the bidder's API callback...
+export const spec = {
+    code: BIDDER_CODE,
+    supportedMediaTypes: ['video'],
+    ...
+}
+
+{% endhighlight %}
+
+### Step 2: Accept video parameters and pass them to your server
 
-// the bidder API's ad response unit that has info like CPM, creative content
-var adUnit;
+See the [AppNexus AST adapter]({{site.baseurl}}/dev-docs/bidders.html#appnexusAst) for an example of how
```

Comment:
I think we need to explain this instead of just pointing to the `appnexusAst` adapter.

Also, the `appnexusAst` adapter will not exist in Prebid 1.0.

Reply:
Good catch, updated this in 973c145 with the below - I tried to extrapolate from our AST adapter's design, let me know if you think it needs tweaking.

<blockquote>

Video parameters are often passed in from the ad unit in a `video` object.

The design of these parameters may vary depending on what your server-side bidder accepts.  If possible, we recommend using the video parameters in the [OpenRTB specification](https://iabtechlab.com/specifications-guidelines/openrtb/).

For examples of video parameters accepted by different adapters, see [the list of bidders with video demand]({{site.baseurl}}/dev-docs/bidders.html#bidder-video-native).

</blockquote>